enhancements/authentication/custom-scc-preemption-prevention.md

```diff
@@ -0,0 +1,260 @@
+for an SCC in the requested namespace.
+If an SCC is applicable, the pod's manifest is potentially mutated according to the SCC logic.
+
+One can see that custom SCCs can easily overrule existing out of the box SCCs
+if the pod's service account or the requesting user has sufficient privileges to use any of them.
+
+In that case, if a custom SCC has higher priority, is more restrictive, or it's name is lexically ordered before an out of the box SCC,
+any OpenShift out of the box core pod will be mutated according to the custom SCC specified behavior.
+
+This can lead to unknown mutations of core OpenShift workloads.
+Worst case it can render OpenShift core workloads dysfunctional
+if i.e. a custom SCC is more restrictive than a previously applicable less restrictive out of the box SCC that was applicable to that workload.
+
+This proposal includes two suggested protection modes, a preferred "hard protection" and a "soft protection" variant
+which are described below.
+
+We need a mechanism to protect OpenShift core workloads such that they are not
+subject to pod mutation due to user provided custom SCCs.
+
+### User Stories
+
+1. As an OpenShift core developer I want to make sure that my shipped workload
+is guaranteed to be associated with an out of the box SCC.
+
+2. As an OpenShift core developer I want to make sure that my shipped workload
+is not preempted by custom user provided SCCs.
+
+### Goals
+
+1. Protect core OpenShift workloads from being preempted by user provided custom SCCs.
+
+2. Be able to pin core OpenShift workloads to a concrete out of the box SCC.
+
+3. Make this change generally available for end users.
+
+### Non-Goals
+
+## Proposal
+
+### "hard protection" - Requesting a required SCC
+
+This proposal suggests to implement a mechanism which enables OpenShift core workloads to enforce
+an SCC to be pinned to a workload.
+This is accomplished by setting an annotation named `openshift.io/required-scc`
+whose value contains the name of the SCC that is supposed to be pinned.
+
+In order to prevent accidental changes to that annotation,
+admission will fail admission if the `openshift.io/required-scc is being changed or added
+for update events.
+
+If the requested SCC does not exist in the system, admission fails with an error.
+
+If the requested SCC does exist, any other SCC available in the system is ignored
+and the requested SCC is the only one being considered for admission.
+
+If the SCC is applicable to the workload, it gets/is used,
+else admission fails with an error.
+
+Pros:
+- The required and pinned SCC is deterministic.
+- No other SCC in the system is taken into account.
+
+Cons:
+- If the required SCC is not applicable, the workload fails to be created/admitted.
+This can be detected in the existing e2e suite.
+- All core OpenShift workloads need to set that annotation in their deployment artifacts.
```

Comment:
i assume it gets set in the podtemplate within the deployment(or daemonset, or whatever resource), right?

Reply:
correct. i added this information in the sentence now as well.